examples/models/llama2/CMakeLists.txt

```diff
@@ -0,0 +1,226 @@
+#
+
+#
+# llama_executor_runner
+#
+
+cmake_minimum_required(VERSION 3.19)
+
+project(llama_runner_example)
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+endif()
+
+if(NOT PYTHON_EXECUTABLE)
+  set(PYTHON_EXECUTABLE python3)
+endif()
+
+if(NOT FLATC_EXECUTABLE)
+  set(FLATC_EXECUTABLE flatc)
+endif()
+
+# Source root directory for executorch.
+if(NOT EXECUTORCH_ROOT)
+  set(EXECUTORCH_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../../..)
+endif()
+
+# Source root directory for pytorch.
+if(NOT TORCH_ROOT)
+  set(TORCH_ROOT ${EXECUTORCH_ROOT}/third-party/pytorch)
+endif()
+
+add_compile_options("-Wall" "-Werror")
+
+include(${EXECUTORCH_ROOT}/build/Utils.cmake)
+
+set(_common_compile_options -Wno-deprecated-declarations -fPIC)
+
+# Let files say "include <executorch/path/to/header.h>".
+set(_common_include_directories ${EXECUTORCH_ROOT}/..)
+
+# Find prebuilt libraries. executorch package should contain
+# portable_ops_lib, etdump, bundled_program.
+find_package(executorch CONFIG REQUIRED)
+target_include_directories(executorch INTERFACE ${_common_include_directories})
+target_compile_options(executorch INTERFACE -DET_EVENT_TRACER_ENABLED)
+
+find_package(
+  gflags REQUIRED PATHS ${CMAKE_CURRENT_BINARY_DIR}/../../../third-party
+)
+
+# portable_ops_lib
+include(${EXECUTORCH_ROOT}/build/Utils.cmake)
+include(${EXECUTORCH_ROOT}/build/Codegen.cmake)
+gen_selected_ops("" "" "ON")
+generate_bindings_for_kernels(
+  ${EXECUTORCH_ROOT}/kernels/portable/functions.yaml ""
+)
+gen_operators_lib("portable_ops_lib" portable_kernels executorch)
+
+set(llama_runner_libs "-framework Foundation")
+
+#
+# The `_<target>_srcs` lists are defined by including ${EXECUTORCH_SRCS_FILE}.
+#
+set(
+  EXECUTORCH_SRCS_FILE
+  "${CMAKE_CURRENT_BINARY_DIR}/../../../executorch_srcs.cmake"
+)
+
+extract_sources(${EXECUTORCH_SRCS_FILE})
+
+
+add_library(extension_module "${CMAKE_CURRENT_SOURCE_DIR}/../../../extension/module/module.cpp")
+target_include_directories(extension_module PUBLIC ${EXECUTORCH_ROOT}/..)
+
+#add_library(extension_data_loader "${CMAKE_CURRENT_SOURCE_DIR}/../../../extension/data_loader/mmap_data_loader.cpp")
+#target_link_libraries(extension_data_loader executorch)
+
+
+# Install libraries
+install(
+  TARGETS extension_module
+  DESTINATION ${CMAKE_BINARY_DIR}/lib
+  INCLUDES
+  DESTINATION ${_common_include_directories})
+
+
+set(llama_schema_headers ${CMAKE_BINARY_DIR}/../../../schema/include/)
+include(${EXECUTORCH_SRCS_FILE})
+target_include_directories(
+  bundled_program
+  INTERFACE
+  ${CMAKE_CURRENT_BINARY_DIR}/../../../sdk/include
+  ${CMAKE_CURRENT_BINARY_DIR}/../../../sdk/bundled_program
+  ${EXECUTORCH_ROOT}/third-party/flatbuffers/include
+  ${EXECUTORCH_ROOT}/third-party/flatcc/include
+  ${llama_schema_headers}
+)
+
+function(define_common_targets)
+    set(EXPORTED_HEADERS_RUNNER
+        "${CMAKE_CURRENT_SOURCE_DIR}/runner/runner.h"
+        "${CMAKE_CURRENT_SOURCE_DIR}/runner/util.h"
+    )
```

Comment:
Hmm like I mentioned last time, we don't need to specifically include these headers, if you do this: https://github.com/pytorch/executorch/blob/main/CMakeLists.txt#L250

Reply:
tried but got error. not knowing for sure a better way, i put line 108 & 109 directly under `target_sources(runner PRIVATE`.